pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java

```diff
@@ -0,0 +1,69 @@
+package org.broadinstitute.dsm.route.somaticresults;
+
+import static spark.Spark.halt;
+
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.broadinstitute.dsm.model.somatic.result.SomaticResultMetaData;
+import org.broadinstitute.dsm.security.RequestHandler;
+import org.broadinstitute.dsm.service.SomaticResultUploadService;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.util.UserUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+@Slf4j
+public class PostSomaticResultUploadRoute extends RequestHandler {
+    private final SomaticResultUploadService service;
+    private static final Logger logger = LoggerFactory.getLogger(PostSomaticResultUploadRoute.class);
+    protected static final String REQUIRED_ROLE = "upload_ror_file";
+
+    public PostSomaticResultUploadRoute(SomaticResultUploadService service) {
+        this.service = service;
+    }
+
+    @Override
+    protected Object processRequest(Request request, Response response, String userId) throws Exception {
+        Gson gson = new Gson();
+        QueryParamsMap queryParams = request.queryMap();
+        String realm = queryParams.get(RoutePath.REALM).value();
+        String ddpParticipantId = queryParams.get(RoutePath.DDP_PARTICIPANT_ID).value();
+        SomaticResultMetaData somaticResultMetaData = gson.fromJson(request.body(), SomaticResultMetaData.class);
+        if (isValidRequest(realm, ddpParticipantId, userId, somaticResultMetaData)) {
+            return service.authorizeUpload(realm, userId, ddpParticipantId, somaticResultMetaData);
```

Comment:
How will the return of `AuthorizeResult` be processed after it is returned here?

Reply:
It will contain the signedUrl (and related record metadata) so the client can post the pdf into GCS and visualize the entry (or errors) in the list of uploaded files in the tab the user is working in.

Reply:
But it also has error information?

Reply:
It does. AuthorizeResult will return a response so the client can tell if the fileName meets naming requirements, if the mimeType is one in the configured list to support, or if the file exceeds the maximum size allocated.  See AuthorizeResultType.  In cases where this happens, we also send back an object that contains the settings the client should be enforcing (e.g. mime type, max file size).  In an error state, the signed URL will be null and so will the SomaticResultUpload since we're rejecting the file.

Reply:
Ah, so we are using the mimetype. :-)
But these bad requests will all go back with status = 500?

Reply:
We are.  We validate it against the list of supported mime types and will return a 200 with a payload that indicates the mime type isn't supported.

Reply:
This behavior is cloned from DSS' own file upload component.

Reply:
Let's talk directly when you get a chance?